backend/de.metas.manufacturing/src/main/java/de/metas/costing/methods/ManufacturingLastPOCostingMethodHandler.java

```diff
@@ -0,0 +1,273 @@
+import de.metas.costing.MoveCostsResult;
+import de.metas.currency.CurrencyPrecision;
+import de.metas.material.planning.IResourceProductService;
+import de.metas.order.OrderLineId;
+import de.metas.product.ProductId;
+import de.metas.product.ResourceId;
+import de.metas.quantity.Quantity;
+import de.metas.util.Services;
+import lombok.NonNull;
+import org.adempiere.exceptions.AdempiereException;
+import org.eevolution.api.CostCollectorType;
+import org.eevolution.api.IPPCostCollectorBL;
+import org.eevolution.api.IPPOrderCostBL;
+import org.eevolution.api.PPCostCollectorId;
+import org.eevolution.api.PPOrderBOMLineId;
+import org.eevolution.api.PPOrderCosts;
+import org.eevolution.api.PPOrderId;
+import org.eevolution.model.I_PP_Cost_Collector;
+import org.springframework.stereotype.Component;
+
+import java.time.Duration;
+import java.util.Optional;
+import java.util.Set;
+
+/*
+ * #%L
+ * de.metas.adempiere.libero.libero
+ * %%
+ * Copyright (C) 2018 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+@Component
+public class ManufacturingLastPOCostingMethodHandler implements CostingMethodHandler
+{
+	// services
+	private final IPPCostCollectorBL costCollectorsService = Services.get(IPPCostCollectorBL.class);
+	private final IResourceProductService resourceProductService = Services.get(IResourceProductService.class);
+	private final IPPOrderCostBL ppOrderCostsService = Services.get(IPPOrderCostBL.class);
+	private final IAcctSchemaDAO acctSchemasRepo = Services.get(IAcctSchemaDAO.class);
+	//
+	private final CostingMethodHandlerUtils utils;
+
+	private static final ImmutableSet<String> HANDLED_TABLE_NAMES = ImmutableSet.<String>builder()
+			.add(CostingDocumentRef.TABLE_NAME_PP_Cost_Collector)
+			.build();
+
+	public ManufacturingLastPOCostingMethodHandler(
+			@NonNull final CostingMethodHandlerUtils utils)
+	{
+		this.utils = utils;
+	}
+
+	@Override
+	public CostingMethod getCostingMethod()
+	{
+		return CostingMethod.LastPOPrice;
+	}
+
+	@Override
+	public Set<String> getHandledTableNames()
+	{
+		return HANDLED_TABLE_NAMES;
+	}
+
+	@Override
+	public Optional<CostDetailCreateResult> createOrUpdateCost(final CostDetailCreateRequest request)
+	{
+		final PPCostCollectorId costCollectorId = request.getDocumentRef().getCostCollectorId(PPCostCollectorId::ofRepoId);
```

Comment:
pls make sure you are not creating the cost detail multiple times.
see ManufacturingAveragePOCostingMethodHandler.createOrUpdateCost,
i.e.
```java
final CostDetail existingCostDetail = utils.getExistingCostDetail(request).orElse(null);
		if (existingCostDetail != null)
		{
			return Optional.of(utils.toCostDetailCreateResult(existingCostDetail));
		}
```